index.js

```diff
@@ -1,8 +1,5 @@
 'use strict'
 
-const PluginVirtual = require('./src/lib/plugin')
-const Token = require('./src/util/token')
-
-module.exports = PluginVirtual
-module.exports.generatePrefix = Token.prefix
-module.exports.generatePublicKey = Token.publicKey
+const Plugin = require('./src/lib/plugin')
+module.exports = Plugin.bind(null, null)
+module.exports.MakePaymentChannelPlugin = (channel) => (Plugin.bind(null, channel))
```

Comment:
I'm pretty sure this doesn't work.

I think it's equivalent to this example:

```
class SomeClass {
  constructor (opts) {
    this.value = opts.value
  }

  get () {
    return this.value
  }
}

const a = new SomeClass({ value: 10 })
const b = SomeClass.bind(null, { value: 10 })
console.log(a.get())
console.log(b.get())
```
If you run it like that you get `b.get is not a function` and if you run it with another pair of parens after the `b` definition you get `TypeError: Class constructor SomeClass cannot be invoked without 'new'`